src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/ValidationErrors.kt

```diff
@@ -6,3 +6,21 @@ value class ValidationErrors(private val errorMap: MutableMap<String, String>) :
 }
 
 fun singleValidationErrorOf(propertyNameToMessage: Pair<String, String>) = ValidationErrors().apply { this[propertyNameToMessage.first] = propertyNameToMessage.second }
+
+class ValidatedScope<EntityType> {
+  private val validationErrors = ValidationErrors()
+
+  val validationErrorsPresent: Boolean
+    get() = validationErrors.any()
```

Comment:
Can we get the validation errors directly from the scope instead of calling `validationErrorsPresent`. Calling `validationErrors.any()` directly feels more expressive to me and cuts down on boilerplate